python-lib/tagger_formatter.py

```diff
@@ -0,0 +1,443 @@
+        else:
+            self._get_tags_row_per_keyword_category(document, row)
+        if not self.matches:
+            self.output_df = self.output_df.append(empty_row, ignore_index=True)
+            self.df_duplicated_lines = self.df_duplicated_lines.append(row)
+
+    def _get_tags_row_per_document_json(
+        self, match: Span, line_full: dict, sentence: Doc
+    ):
+        """
+        Called by _row_per_document_json on each sentence
+        Returns a dictionary containing precisions about each tag
+        """
+        keyword = match.text
+        tag = self.tagger.keyword_to_tag[keyword]
+        sentence = sentence.text
+        if tag not in line_full.keys():
+            json_dictionary = {
+                "occurence": 1,
+                "sentences": [sentence],
+                "keywords": [keyword],
+            }
+            line_full[tag] = json_dictionary
+        else:
+            line_full[tag]["occurence"] += 1
+            line_full[tag]["sentences"].append(sentence)
+            line_full[tag]["keywords"].append(keyword)
+
+        return line_full
+
+    def _get_tags_row_per_document(
+        self,
+        sentence: Doc,
+        list_matched_tags: List,
+        tag_columns_in_document: Dict[AnyStr, List],
+        string_sentence: AnyStr,
+        string_keywords: AnyStr,
+    ):
+        """
+        Called by _row_per_document on each sentence
+        Returns the tags, sentences and keywords linked to the given sentence
+        """
+        tag_columns_in_sentence = defaultdict(list)
+        matches = self.tagger.matcher_dict[self.tagger.language](
+            sentence, as_spans=True
+        )
+        for match in matches:
+            keyword = match.text
+            tag = self.tagger.keyword_to_tag[keyword]
+            if tag not in list_matched_tags:
+                list_matched_tags.append(tag)
+                tag_columns_in_sentence[self.output_dataset_columns[2]].append(tag)
+            string_keywords = string_keywords + " " + keyword
+            string_sentence = string_sentence + sentence.text
+
+        if tag_columns_in_sentence != {}:
+            tag_columns_in_document[self.output_dataset_columns[2]].extend(
+                tag_columns_in_sentence[self.output_dataset_columns[2]]
+            )
+        return tag_columns_in_document, string_sentence, string_keywords
+
+    def _get_tags_row_per_document_category(
+        self, match: Span, line: dict, line_full: dict, sentence: Doc
+    ):
+        """
+        Called by _row_per_document_category
+        Writes the needed informations about founded tags:
+        -line is a dictionary {category:tag}
+        -line_full is a dictionary containing full information about the founded tags
+        """
+        keyword = match.text
+        tag = self.tagger.keyword_to_tag[keyword]
+        category = match.label_
+        sentence = sentence.text
+        if tag not in line_full[category]:
+            json_dictionary = {
```

Comment:
Note on variable naming, be careful not to mix the variable name with its type. In this case, your variable is not a JSON. You get a JSON when you serialize a dictionary instance. 
On the need to indicate the variable type in its name, it can sometimes help make the code readable. But most of the time, it's not a good practice, so it requires thoughts. A good variable name should make the "purpose" of the variable clear.
In your case, you could either assign the dictionary directly to `line_full[category][tag]`, or call it something like `tag_count`